src/app/calculator/furnaces/flue-gas/flue-gas-form-volume/flue-gas-form-volume.component.ts

```diff
@@ -0,0 +1,286 @@
+import { FlueGasService } from '../flue-gas.service';
+
+@Component({
+  selector: 'app-flue-gas-form-volume',
+  templateUrl: './flue-gas-form-volume.component.html',
+  styleUrls: ['./flue-gas-form-volume.component.css']
+})
+export class FlueGasFormVolumeComponent implements OnInit, OnDestroy {
+
+  @Input()
+  settings: Settings;
+  @Input()
+  isBaseline: boolean;
+  @Input()
+  selected: boolean;
+  @Input()
+  inModal: boolean;
+  @ViewChild('formElement', { static: false }) formElement: ElementRef;
+
+  @ViewChild('materialModal', { static: false }) public materialModal: ModalDirective;
+
+  resetDataSub: Subscription;
+  generateExampleSub: Subscription;
+  coolingTowerOutputSub: Subscription;
+  modificationDataSub: Subscription;
+
+  byVolumeForm: FormGroup;
+
+  options: any;
+  calculationMethods: Array<string> = [
+    'Excess Air',
+    'Oxygen in Flue Gas'
+  ];
+  calculationExcessAir: number = 0.0;
+  calculationFlueGasO2: number = 0.0;
+  calcMethodExcessAir: boolean;
+  flueTemperatureWarning: boolean = false;
+  tempMin: number;
+  warnings: FlueGasWarnings;
+
+  constructor(private flueGasService: FlueGasService, 
+              private phastService: PhastService, 
+              private suiteDbService: SuiteDbService,
+              private convertUnitsService: ConvertUnitsService) {
+  }
+
+  ngOnInit() {
+    this.options = this.suiteDbService.selectGasFlueGasMaterials();
+    this.initSubscriptions();
+  }
+
+  ngOnDestroy() {
+    this.resetDataSub.unsubscribe();
+    this.generateExampleSub.unsubscribe();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.selected && !changes.selected.firstChange) {
+      if (this.selected == false) {
+        this.byVolumeForm.disable();
+      } else {
+        this.byVolumeForm.enable();
+      }
+    }
+  }
+
+  initSubscriptions() {
+    this.resetDataSub = this.flueGasService.resetData.subscribe(value => {
+      this.setForm();
+      })
+    this.generateExampleSub = this.flueGasService.generateExample.subscribe(value => {
+      this.setForm();
+    })
+  }
+
+  initFormSetup() {
+    if (this.selected == false) {
+      this.byVolumeForm.disable();
+    }
+    if (this.byVolumeForm.controls.gasTypeId.value && this.byVolumeForm.controls.gasTypeId.value !== '') {
+      if (this.byVolumeForm.controls.CH4.value === '' || !this.byVolumeForm.controls.CH4.value) {
+        this.setProperties();
+      }
+    }
+    this.setCalcMethod();
+    this.calcExcessAir();
+    this.setCombustionValidation();
+    this.setFuelTempValidation();
+    this.tempMin = 212;
+    this.tempMin = this.convertUnitsService.value(this.tempMin).from('F').to(this.settings.steamTemperatureMeasurement);
+    this.tempMin = this.convertUnitsService.roundVal(this.tempMin, 1);
+    this.calculate();
+  }
+
+  setForm() {
+    let updatedFlueGasData: FlueGas;
+    if (this.isBaseline) {
+      let currentBaseline: FlueGas = this.flueGasService.baselineData.getValue();
```

Comment:
don't need the "currentBaseline" and "currentModification" variables. Can just set the "updatedFlueFasData" directly